### PR TITLE
Update to KiCad 5.0.0

### PR DESCRIPTION
**Pull request by lkundrak**
Do not merge -- this is essentially just intended to gather feeback and testing by the bot (thanks @nedrichards)

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build

**Comment by jmaibaum** [comment]
Hi, sorry for chiming in here.

I've got just a quick question: Is there anything that technically prevents building with `-DKICAD_SPICE=ON`? Or is this simply something that has not yet been tested enough?

**Comment by lkundrak** [comment]
@jmaibaum wrote:
> I've got just a quick question: Is there anything that technically prevents building with > -DKICAD_SPICE=ON? Or is this simply something that has not yet been tested enough?

No particular reason. This was in fact just copied from Fedora SPEC, but there's no good reason there either now that a recent enough ngspice is there.

I'll eventually look into enabling it. Of course, sending in patches (or beer, for that matter) my way may speed things up.

**Comment by lkundrak** [comment]
@jmaibaum you owe me a beer now. Alternatively, please let me know if the ngspice support works for you.

**Comment by lkundrak** [comment]
@nedrichards i'm wondering if you could poke the bot so that it does a test build of the new version? Thanks!

**Comment by nedrichards** [comment]
bot, build

**Comment by nedrichards** [comment]
Poked - https://flathub.org/builds/#/builders/2/builds/2839

File: org.kicad_pcb.KiCad.json [hunk 1]
```diff
@@ -39,36 +40,74 @@
...
                 "./bootstrap.sh --prefix=/app",
-                "./b2 install --with-context --with-date_time --with-filesystem --with-iostreams --with-locale --with-program_options --with-regex --with-system --with-thread --prefix=/app"
+                "./b2 install --with-context --with-date_time --with-filesystem --with-iostreams --with-locale --with-program_options --with-regex --with-system --with-thread --with-test --prefix=/app"
             ]
         },
         {
-            "name": "kicad",
+            "name": "ngspice",
+            "rm-configure": true,
+            "config-opts": [ "--enable-shared", "--disable-static", "--without-x", "--disable-silent-rules" ],
```
Comment (nedrichards) [inline org.kicad_pcb.KiCad.json]: looks like this also needs "--with-ngshared" according to https://flathub.org/builds/#/builders/26/builds/343

Comment (jmaibaum) [reply org.kicad_pcb.KiCad.json]: The `get_libngspice_so.sh` script also mentioned in the log uses the following command to configure ngspice (see https://github.com/KiCad/kicad-source-mirror/blob/master/scripts/get_libngspice_so.sh):

`./configure --with-ngshared --enable-xspice --enable-cider ${CFG_OPTIONS}`

where `${CFG_OPTIONS}` expands to `--enable-openmp` on linux platforms.

**Pull request merged**